pkg/controller/backup_controller.go

```diff
@@ -281,6 +286,53 @@ func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackup
 	return backupLocation, validationErrors
 }
 
+// defaultAndValidateSnapshotLocations ensures:
+// - each location name in Spec VolumeSnapshotLocation exists as a location
+// - exactly 1 location per existing or default provider
+// - a given default provider's location name is added to the Spec VolumeSnapshotLocation if it does not exist as a VSL
+func (c *backupController) defaultAndValidateSnapshotLocations(itm *api.Backup, defaultLocations map[string]string) []string {
+	var errors []string
+	perProviderLocationName := make(map[string]string)
+	var finalLocationNameList []string
+	for _, locationName := range itm.Spec.VolumeSnapshotLocations {
+		// validate each locationName exists as a VolumeSnapshotLocation
+		location, err := c.snapshotLocationLister.VolumeSnapshotLocations(itm.Namespace).Get(locationName)
+		if err != nil {
+			errors = append(errors, fmt.Sprintf("error getting volume snapshot location named %s: %v", locationName, err))
+			continue
+		}
+
+		// ensure we end up with exactly 1 locationName *per provider*
+		providerLocationName := perProviderLocationName[location.Spec.Provider]
+		if providerLocationName != "" {
+			// if > 1 location name per provider as in ["aws-us-east-1" | "aws-us-west-1"] (same provider, multiple names)
+			if providerLocationName != locationName {
+				errors = append(errors, fmt.Sprintf("more than one VolumeSnapshotLocation specified for provider %s: %s", location.Spec.Provider, locationName))
```

Comment:
We should probably also include `providerLocationName` in this error msg to be clear about what the dupe for the provider is.  Picked up on this from looking at the unit test msgs.